File: app/src/main/java/com/vsa/filmoteca/data/repository/database/WidgetDataBaseHelper.java
package com.vsa.filmoteca.data.repository.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteDatabase.CursorFactory;
import android.database.sqlite.SQLiteOpenHelper;

public class WidgetDataBaseHelper extends SQLiteOpenHelper{

    public static final String DATABASE_NAME = "WidgetDatabase.db";
    public static final int DATABASE_VERSION = 2;

    public static final String TABLE_MOVIES = "movies";

	//Sentencia SQL para crear la tabla de Usuarios
	String sqlCreate = "CREATE TABLE " + MoviesTable.TABLE_NAME + " (cod INTEGER, " +
            MoviesTable.COLUMN_TITLE + " TEXT, " +
			MoviesTable.COLUMN_SUBTITLE + " TEXT, " +
            MoviesTable.COLUMN_DESCRIPTION + " TEXT, " +
            MoviesTable.COLUMN_DATE + " TEXT, " +
            MoviesTable.COLUMN_URL + " TEXT)";

    public WidgetDataBaseHelper(Context context) {
        this(context, DATABASE_NAME, null, DATABASE_VERSION);
    }
	
	public WidgetDataBaseHelper(Context context, String name, CursorFactory factory,
                                int version) {
		super(context, name, factory, version);
	}

	@Override
	public void onCreate(SQLiteDatabase db) {
		// TODO Auto-generated method stub
		db.execSQL(sqlCreate);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int arg1, int arg2) {
		// TODO Auto-generated method stub
		//Se elimina la versión anterior de la tabla
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_MOVIES);
 
        //Se crea la nueva versión de la tabla
        db.execSQL(sqlCreate);
	}

}

File: app/src/main/java/com/vsa/filmoteca/data/repository/database/WidgetDataBaseHelper.kt
package com.vsa.filmoteca.data.repository.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteDatabase.CursorFactory
import android.database.sqlite.SQLiteOpenHelper

class WidgetDataBaseHelper @JvmOverloads constructor(context: Context, name: String = DATABASE_NAME, factory: CursorFactory? = null,
                                                     version: Int = DATABASE_VERSION) : SQLiteOpenHelper(context, name, factory, version) {

    private var sqlQueryCreateUsersTable = "CREATE TABLE " + MoviesTable.TABLE_NAME + " (cod INTEGER, " +
            MoviesTable.COLUMN_TITLE + " TEXT, " +
            MoviesTable.COLUMN_SUBTITLE + " TEXT, " +
            MoviesTable.COLUMN_DESCRIPTION + " TEXT, " +
            MoviesTable.COLUMN_DATE + " TEXT, " +
            MoviesTable.COLUMN_URL + " TEXT)"

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(sqlQueryCreateUsersTable)
    }

    override fun onUpgrade(db: SQLiteDatabase, arg1: Int, arg2: Int) {
        db.execSQL("DROP TABLE IF EXISTS $TABLE_MOVIES")
        db.execSQL(sqlQueryCreateUsersTable)
    }

    companion object {

        const val DATABASE_NAME = "WidgetDatabase.db"
        const val DATABASE_VERSION = 2

        const val TABLE_MOVIES = "movies"

    }

}
